File: api/src/main/java/com/hirrao/javaee/mapper/BlogMapper.java
package com.hirrao.javaee.mapper;

import com.hirrao.javaee.entity.Blog;
import com.baomidou.mybatisplus.core.mapper.BaseMapper;
import org.apache.ibatis.annotations.*;

import java.util.List;

@Mapper
public interface BlogMapper extends BaseMapper<Blog> {
    @Select("select * from blog where blogId = #{blogId}")
    Blog findByBlogId(Long blogId);

    @Select("select * from blog where uid = #{uid}")
    List<Blog> findByUid(Long uid);

    @Select("select * from blog where title = #{title}")
    List<Blog> findByTitle(String title);

    @Insert("insert into blog(blogId, content, createTime, updateTime, uid, title) values(#{blogId}, #{content}, #{createTime}, #{updateTime}, #{uid}, #{title})")
    void addBlog(Long blogId, String content, String createTime, String updateTime, Long uid, String title);

    @Update("update blog set content=#{content}, updateTime=#{updateTime}, title=#{title} where blogId=#{blogId}")
    void update(Long blogId, String content, String updateTime, String title);

    @Delete("DELETE FROM blog WHERE blogId = #{blogId}")
    void delete(Long blogId);
}

File: api/src/main/kotlin/com/hirrao/javaee/mapper/BlogMapper.kt
package com.hirrao.javaee.mapper

import com.hirrao.javaee.entity.Blog
import com.baomidou.mybatisplus.core.mapper.BaseMapper
import org.apache.ibatis.annotations.*

@Mapper
interface BlogMapper : BaseMapper<Blog?> {
    @Select("select * from blog where blogId = #{blogId}")
    fun findByBlogId(blogId: Long?): Blog?

    @Select("select * from blog where uid = #{uid}")
    fun findByUid(uid: Int): List<Blog?>?

    @Select("select * from blog where title = #{title}")
    fun findByTitle(title: String?): List<Blog?>?

    @Insert("insert into blog(blogId, content, createTime, updateTime, uid, title) values(#{blogId}, #{content}, #{createTime}, #{updateTime}, #{uid}, #{title})")
    fun addBlog(blogId: Long?, content: String?, createTime: String?, updateTime: String?, uid: Int, title: String?)

    @Update("update blog set content=#{content}, updateTime=#{updateTime}, title=#{title} where blogId=#{blogId}")
    fun update(blogId: Long?, content: String?, updateTime: String?, title: String?)

    @Delete("DELETE FROM blog WHERE blogId = #{blogId}")
    fun delete(blogId: Long?)
}
